### PR TITLE
Added binary search topic in DSA

### PR DESCRIPTION
**Pull request by Abhii-07**
- [x] Added binary search to DSA with proper details

### CODE REVIEW

File: DSA/binary_search.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
+// Binary Search
+
+// Explaination : 
+// Binary Search is applied to a sorted array or list. It's a divide-and-conquer algorithm that efficiently finds the position of a target element in the array. 
+// Binary Search is a highly efficient algorithm for searching in sorted data, with a time complexity of O(log n), where n is the size of the array.
+
+function binarySearch(arr, target) {
```
Comment (ishwarrimal) [inline DSA/binary_search.js:11]: Can you please follow JS Doc standards for comments?
https://jsdoc.app/about-getting-started.html

**Review by ishwarrimal — COMMENTED** [review]
We've got to find a way to make this simpler. 
Adding code will not be very helpful, people can find code anywhere.
My suggestion is to add 'pseudo code' and some explaination (maybe make use of chatgpt) along with the code.

Thoughts?

File: DSA/binary_search.js (hunk 1 above)
Comment (Abhii-07) [reply DSA/binary_search.js:11]: Sure,  I'll work on it asap

**Comment by Abhii-07** [comment]
I will try putting a  better explanation.

**Comment by Abhii-07** [comment]
I have updated comments, not sure about better explanation. Maybe I will work on it later, but for now I guess you can merge this. Please check.

**Review by ishwarrimal — APPROVED** [review]

**Pull request merged**